app/javascript/app/document-capture/components/acuant-capture.jsx

```diff
@@ -1,46 +1,80 @@
 import React, { useContext, useState } from 'react';
+import PropTypes from 'prop-types';
 import AcuantContext from '../context/acuant';
 import AcuantCaptureCanvas from './acuant-capture-canvas';
+import FileInput from './file-input';
 import FullScreen from './full-screen';
+import Button from './button';
 import useI18n from '../hooks/use-i18n';
+import DeviceContext from '../context/device';
+import DataURLFile from '../models/data-url-file';
 
-function AcuantCapture() {
-  const { isReady, isError } = useContext(AcuantContext);
+function AcuantCapture({ label, hint, bannerText, value, onChange, className }) {
+  const { isReady, isError, isCameraSupported } = useContext(AcuantContext);
   const [isCapturing, setIsCapturing] = useState(false);
-  const [capture, setCapture] = useState(null);
+  const { isMobile } = useContext(DeviceContext);
   const { t } = useI18n();
+  const hasCapture = !isError && (isReady ? isCameraSupported : isMobile);
 
-  if (isError) {
-    return 'Error!';
-  }
-
-  if (!isReady) {
-    return 'Loading…';
-  }
-
-  if (capture) {
-    const { data, width, height } = capture.image;
-    return <img alt="Captured result" src={data} width={width} height={height} />;
+  let startCaptureIfSupported;
+  if (hasCapture) {
+    startCaptureIfSupported = (event) => {
+      event.preventDefault();
+      setIsCapturing(true);
+    };
   }
 
   return (
-    <>
+    <div className={className}>
       {isCapturing && (
         <FullScreen onRequestClose={() => setIsCapturing(false)}>
           <AcuantCaptureCanvas
             onImageCaptureSuccess={(nextCapture) => {
-              setCapture(nextCapture);
+              onChange(nextCapture.image.data);
```

Comment:
do we ever need to check that `image` is null?

Reply:
> do we ever need to check that `image` is null?

It might not hurt to be safe, but as best I can tell, we should be able to rely on it being assigned.

Line 1233 of [the SDK source](https://github.com/Acuant/JavascriptWebSDKV11/blob/master/SimpleHTMLApp/webSdk/dist/AcuantJavascriptWebSdk.js):

```js
let imgData = getImageData()
croppedResult = {
    image: { 
        data: imgData,
        width,
        height
    }, 
    cardType,
    dpi: calculateDpi(width, height, cardType == 2)
};
```

Reply:
On the topic of types, even in absence of Acuant shipping their own types, it could be wise for us to [define our own](https://jsdoc.app/tags-typedef.html), which could at least help us keep ourselves accountable to our own understanding of the shape of the object we're expecting to receive.

Reply:
Added types in aef63a4.

![image](https://user-images.githubusercontent.com/1779930/89685601-a915e800-d8ca-11ea-8ceb-6c846899cf41.png)